### PR TITLE
Upgrades

### PR DESCRIPTION
**Pull request by bhollis**
A bunch of big upgrades here:

1. ESLint changes - most of the new rules are disabled. Some because they're incompatible with how we're operating (e.g. complaining about comparing an enum to `number`) and some should be enabled later.
2. Prettier changes - general formatting, and they've enabled trailing commas. I'll switch to the "all trailing commas everywhere" config in a separate PR.
3. Switch from pretty-quick to lint-staged because it works. As a side-effect this runs more lints as part of pre-commit. I'm not sure if that's good - ~if commits are slow for you all, we should disable eslint/stylelint~ these are too slow so I disabled the eslint one.
4. Upgrade downshift, which changes behavior on focusing the search bar. The search dropdown no longer opens on focus, but does on click. I think it's OK.
5. Switched from the abandoned `babel-plugin-lodash` to a fork that is more maintained, so it doesn't complain all the time now.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -288,6 +289,30 @@
...
+          "number": false,
+          "string": true
+        }
+      }
+    ],
+    "@typescript-eslint/no-unsafe-assignment": "off",
+    "@typescript-eslint/no-unsafe-call": "off",
+    "@typescript-eslint/no-unsafe-member-access": "off",
+    "@typescript-eslint/no-unsafe-return": "off",
+    "@typescript-eslint/unbound-method": "off",
```
Comment (robojumper) [inline .eslintrc:314]: If it weren't for it flagging lodash's stubs/noop, this one wouldn't be that bad. It does flag

https://github.com/DestinyItemManager/DIM/blob/5fdc84262ab820416136f7695a1b5cafd334009e/src/app/stream-deck/async-module.ts#L239

which in my limited understanding seems like it catches a bug here?

**Review by robojumper — APPROVED** [review]
Doesn't seem to make nodemon work reliably for me, but the nodemon part also isn't worse than before 🤷 

👍 for trailing commas

**Comment by bhollis** [comment]
Happy to switch to something other than nodemon if it works better - I haven't seen any problems on macOS.

File: .eslintrc (hunk 1 above)
Comment (bhollis) [reply .eslintrc:314]: Yeah I think I'll swap out noop for our own version with proper types.

**Comment by robojumper** [comment]
> Happy to switch to something other than nodemon if it works better - I haven't seen any problems on macOS

I assume a requirement should be that it's installable from the npm registry? I'm currently testing [`watchexec`](https://github.com/watchexec/watchexec) with a VS Code user task and it seems to work a lot better for me.

**Comment by bhollis** [comment]
Yeah, it would be nice to not require a separate setup step

**Pull request merged**